### PR TITLE
Extend Order's Product-Proposal

### PR DESCRIPTION
**Pull request by fp-ps**

### CODE REVIEW

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/adprocess/ADProcessPostProcessService.java [hunk 1]
```diff
@@ -372,7 +372,7 @@ else if (ViewOpenTarget.NewBrowserTab.equals(target))
 				return OpenViewAction.builder()
 						.viewId(ViewId.ofViewIdString(viewToOpen.getViewId()))
 						.profileId(ViewProfileId.fromJson(viewToOpen.getProfileId()))
-						.targetTab(recordsToOpen != null ? recordsToOpen.getTargetTab() : RecordsToOpen.TargetTab.SAME_TAB_OVERLAY)
+						.targetTab(recordsToOpen != null ? recordsToOpen.getTargetTab() : RecordsToOpen.TargetTab.NEW_TAB)
```
Comment (metas-ts) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/adprocess/ADProcessPostProcessService.java:375]: this change made me a bit nervous, but checking the code it seems clear that you are fixing an old bug from commit  https://github.com/metasfresh/metasfresh/commit/b2e0d90701f3a66138bd225aaa13d40ad31779bf 🙏

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/order/products_proposal/model/ProductsProposalRow.java [hunk 2]
```diff
@@ -100,8 +102,13 @@ public static ProductsProposalRow cast(final IViewRow row)
 	@ViewColumn(seqNo = 70, fieldName = FIELD_IsCampaignPrice, captionKey = "IsCampaignPrice", widgetType = DocumentFieldWidgetType.YesNo)
 	private final boolean isCampaignPrice;
 
+	/**
```
Comment (metas-ts) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/order/products_proposal/model/ProductsProposalRow.java:105]: didn't this work with `ViewColumnHelper` and `Displayed.SYSCONFIG`?

**Review by metas-ts — APPROVED** [review]
lgtm, but please check my questions. depending on the outcomes it might make sense to do a little follow-up

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/order/products_proposal/model/ProductsProposalRow.java (hunk 2 above)
Comment (cp-ps) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/order/products_proposal/model/ProductsProposalRow.java:105]: nope, as the ProductsProposalRow is shared between multiple modals so the fields are programmatically added via `de.metas.ui.web.view.descriptor.ViewLayout.Builder#addElementsFromViewRowClassAndFieldNames`.. a solution would be to refactor everything such that every modal gets its own row definition.

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java [hunk 3]
```diff
@@ -41,6 +42,8 @@ public interface IADWindowDAO extends ISingletonService
 
 	AdWindowId getWindowIdByInternalName(String internalName);
 
+	Optional<AdWindowId> getOverridingWindowId(AdWindowId adWindowId);
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java:45]: this is not needed, pls read below

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/impl/ADWindowDAO.java [hunk 4]
```diff
@@ -111,6 +111,20 @@ public AdWindowId getWindowIdByInternalName(@NonNull final String internalName)
 		return windowIdsByInternalName.getOrLoad(internalName, this::retrieveWindowIdByInternalName);
 	}
 
+	@Override
+	@NonNull
+	public Optional<AdWindowId> getOverridingWindowId(@NonNull final AdWindowId overridesWindowId)
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/impl/ADWindowDAO.java:116]: not needed

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java (hunk 3 above)
Comment (teosarca) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java:45]: wherever you need to get the actual window ID, pls use `de.metas.document.references.zoom_into.CustomizedWindowInfoMapRepository`

e.g.
```java
		return customizedWindowInfoMapRepository.get()
				.getCustomizedWindowInfo(adWindowId)
				.map(CustomizedWindowInfo::getCustomizationWindowId)
				.orElse(adWindowId);
```

That code snippet will give you the effective window ID to be used.

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/order/products_proposal/process/WEBUI_ProductsProposal_ZoomToQuotations.java [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+				.builder(windowId)
+				.setParentViewId(parentViewId)
+				.addStickyFilters(filter)
+				.build();
+	}
+
+	@NonNull
+	private AdWindowId getQuotationWindowId()
+	{
+		final AdWindowId salesWindowId = windowDAO.getWindowIdByInternalName(WINDOW_SalesOrder_InternalName);
```
[the remark this answers is not in the document]
Comment (teosarca) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/order/products_proposal/process/WEBUI_ProductsProposal_ZoomToQuotations.java:145]: @metas-ts  cc @fp-ps see https://github.com/metasfresh/metasfresh/pull/13511/files#r956989019

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java (hunk 3 above)
Comment (teosarca) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java:45]: => https://github.com/metasfresh/metasfresh/pull/13514

**Pull request merged**